### DISCUSSION TITLE
How to define model spaces in Keras?

### DISCUSSION THREAD

**Discussion by sayakpaul** [microsoft/nni#3875] [Q&A] [answered]
I referred to [this document](https://nni.readthedocs.io/en/stable/NAS/QuickStart.html#define-your-model-space) to get a handle on how to define a model space but it seems to be specific for PyTorch and uses PyTorch Lightning for training. My questions are:

* What if I wanted to use the native PyTorch-style training? 
* How do I extend these to Keras?

**Comment by ultmaster** [comment, marked as answer]
* Functional training: https://nni.readthedocs.io/en/stable/NAS/WriteTrainer.html#with-functionalevaluator
* Keras is currently not supported yet. tf/keras is current in plan and will be supported in future releases.

**Reply by sayakpaul** [reply]
Okay got it.
